src/core/provider.h

```diff
@@ -34,6 +34,37 @@
 namespace nvidia { namespace inferenceserver {
 
 class InferenceBackend;
+class SystemMemoryInferResponseProvider;
+
+//
+// Memory block used for accessing / storing data for providers
+// Note:
+//   - It is only responsible for deallocating memory obtained by ReserveBlock()
+//   - The functions are not thread-safe
+//
+class SystemMemoryBlock {
+ public:
+  // Create an SystemMemoryBlock object with 'byte_size'
+  SystemMemoryBlock(size_t byte_size);
+
+  // Create an SystemMemoryBlock that points to 'buffer' provided.
+  // 'buffer' is the memory that has been allocated. For instance,
+  // content returned by InferRequestProvider.GetNextInputContent().
+  // 'byte_size' is the size of the buffer
+  SystemMemoryBlock(const void* buffer, size_t byte_size);
```

Comment:
I still don't understand the use case where this constructor is used. GetNextInputContent is used only by the backends to read input. The request provider content also has a lifetime tied to the provider, so you can't take a reference to it.